nni/compression/pytorch/speedup/compressor.py

```diff
@@ -148,40 +341,128 @@ def replace_compressed_modules(self):
+                    '.')[-1], compressed_module)
+            elif reindex_dim is not None and reindex is not None:
+                # reindex the output of this submodule and replace the orginal module
+                new_submodule = ReindexModule(
+                    compressed_module, reindex_dim, reindex)
+                setattr(super_module, g_node.name.split(
+                    '.')[-1], new_submodule)
+            return new_submodule
+        elif g_node.type == 'func':
+            _logger.info("Warning: cannot replace (name: %s, op_type: %s) which is func type",
+                         unique_name, g_node.op_type)
+            return None
+        else:
+            raise RuntimeError("Unsupported node type: {}".format(g_node.type))
+
+    def initialize_speedup(self):
+        """
+        Do some initial work for speedup.
+        """
+        # initialize the self.debugname_to_value
+        # build a mapping table from the debug name of the tensor
+        # to its value node in the graph
+        traced_graph = self.torch_graph.trace.graph
+        for node in traced_graph.nodes():
+            for _input in node.inputs():
+                debug_name = _input.debugName()
+                if debug_name not in self.debugname_to_value:
+                    self.debugname_to_value[debug_name] = _input
+            for _output in node.outputs():
+                debug_name = _output.debugName()
+                if debug_name not in self.debugname_to_value:
+                    self.debugname_to_value[debug_name] = _output
+        # put the model itself into internel_result to perform the
+        # value inference for the 'prim::GetAttr', the first ClassType
+        # of the whole graph is the model class
+
+        for graph_input in traced_graph.inputs():
+            if graph_input.type().kind() == 'ClassType':
+                self.internal_result[graph_input.debugName()] = self.bound_model
+                break
 
     def speedup_model(self):
         """
-        There are basically two steps:
-        first, do mask/shape inference,
-        second, replace modules
+        There are basically two steps: first, do mask/shape inference,
+        second, replace modules.
         """
-        training = self.bound_model.training
-        _logger.info("start to speed up the model")
 
-        _logger.info("fix the mask conflict of the interdependent layers")
-        _, conv_prune_dim = fix_mask_conflict(self.masks, self.bound_model, self.dummy_input)
-        set_conv_prune_dim(conv_prune_dim)
+        _logger.info("start to speed up the model")
+        self.initialize_speedup()
+        training = self.bound_model.training
+        # set to the evaluation mode
+        self.bound_model.train(False)
+        # TODO suppose to fix the conflict after the sparsity propagation
+        # which is more elegent
+        fix_mask_conflict(self.masks, self.bound_model, self.dummy_input)
```

Comment:
do we still need this function?

Reply:
Yes. Basically, we have two ways to resolve the mask conflict:(1) `fix_mask_conflict` (2) `padding/reindex the module`(currently removed from this pr). With the second way(in the future), we may achieve an automatic conflict fixing without analyzing the model architecture.